File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/BooleanUniform.java
package org.saar.lwjgl.opengl.shaders.uniforms;

import org.lwjgl.opengl.GL20;

public abstract class BooleanUniform extends UniformBase implements Uniform {

    @Override
    public final void doLoad() {
        final int v = getUniformValue() ? 1 : 0;
        GL20.glUniform1i(getLocation(), v);
    }

    public abstract boolean getUniformValue();
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shader/uniforms/BooleanUniform.kt
package org.saar.lwjgl.opengl.shader.uniforms

import org.lwjgl.opengl.GL20

abstract class BooleanUniform : Uniform {

    final override fun load(location: Int) {
        val value: Boolean = this.value
        GL20.glUniform1i(location, if (value) 1 else 0)
    }

    abstract val value: Boolean
}